test/cli_ps_test.go

```diff
@@ -71,6 +71,124 @@ func (suite *PouchPsSuite) TestPsWorks(c *check.C) {
+	c.Assert(exist2, check.Equals, false)
+	c.Assert(exist3, check.Equals, false)
+
+	res = command.PouchRun("ps", "-f", "label=a=b", "-a").Assert(c, icmd.Success)
+	kv = psToKV(res.Combined())
+	_, exist1 = kv[labelA]
+	_, exist2 = kv[labelB]
+	_, exist3 = kv[name]
+	c.Assert(exist1, check.Equals, true)
+	c.Assert(exist2, check.Equals, false)
+	c.Assert(exist3, check.Equals, true)
+
+	res = command.PouchRun("ps", "-f", "status=running").Assert(c, icmd.Success)
+	kv = psToKV(res.Combined())
+	_, exist1 = kv[labelA]
+	_, exist2 = kv[labelB]
+	_, exist3 = kv[name]
+	c.Assert(exist1, check.Equals, true)
+	c.Assert(exist2, check.Equals, true)
+	c.Assert(exist3, check.Equals, false)
+
+	res = command.PouchRun("ps", "-f", "name="+labelA).Assert(c, icmd.Success)
+	kv = psToKV(res.Combined())
+	_, exist1 = kv[labelA]
+	_, exist2 = kv[labelB]
+	_, exist3 = kv[name]
+	c.Assert(exist1, check.Equals, true)
+	c.Assert(exist2, check.Equals, false)
+	c.Assert(exist3, check.Equals, false)
+}
+
+// TestPsFilterUnequal tests "pouch ps -f" filter unequal condition work
+func (suite *PouchPsSuite) TestPsFilterUnequal(c *check.C) {
+	labelA := "unequal-label-a"
+	command.PouchRun("run", "-d", "--name", labelA, "-l", "a=b", busyboxImage, "top").Assert(c, icmd.Success)
+	defer DelContainerForceMultyTime(c, labelA)
+
+	labelB := "unequal-label-b"
+	command.PouchRun("run", "-d", "--name", labelB, "-l", "b=c", busyboxImage, "top").Assert(c, icmd.Success)
+	defer DelContainerForceMultyTime(c, labelB)
+
+	labelC := "unequal-label-c"
+	command.PouchRun("run", "-d", "--name", labelC, "-l", "a=c", busyboxImage, "top").Assert(c, icmd.Success)
+	defer DelContainerForceMultyTime(c, labelC)
+
+	name := "non-running-lable-a"
+	command.PouchRun("create", "--name", name, "-l", "a=c", busyboxImage, "top").Assert(c, icmd.Success)
+	defer DelContainerForceMultyTime(c, name)
+
+	res := command.PouchRun("ps", "-f", "label=b!=c").Assert(c, icmd.Success)
+	kv := psToKV(res.Combined())
+	_, exist1 := kv[labelA]
+	_, exist2 := kv[labelB]
+	_, exist3 := kv[name]
+	_, exist4 := kv[labelC]
+	c.Assert(exist1, check.Equals, false)
+	c.Assert(exist2, check.Equals, false)
+	c.Assert(exist3, check.Equals, false)
+	c.Assert(exist4, check.Equals, false)
+
+	res = command.PouchRun("ps", "-f", "label=a!=b").Assert(c, icmd.Success)
+	kv = psToKV(res.Combined())
+	_, exist1 = kv[labelA]
+	_, exist2 = kv[labelB]
+	_, exist3 = kv[name]
+	_, exist4 = kv[labelC]
+	c.Assert(exist1, check.Equals, false)
+	c.Assert(exist2, check.Equals, false)
+	c.Assert(exist3, check.Equals, false)
+	c.Assert(exist4, check.Equals, true)
+
+	res = command.PouchRun("ps", "-f", "label=a!=b", "-a").Assert(c, icmd.Success)
+	kv = psToKV(res.Combined())
+	_, exist1 = kv[labelA]
+	_, exist2 = kv[labelB]
+	_, exist3 = kv[name]
+	_, exist4 = kv[labelC]
+	c.Assert(exist1, check.Equals, false)
+	c.Assert(exist2, check.Equals, false)
+	c.Assert(exist3, check.Equals, true)
+	c.Assert(exist4, check.Equals, true)
+}
+
```

Comment:
could also add negative tests, like:
-f  null
-f wrong format

and use multiple -f

Reply:
All case has add.